drivers/rtc/rtc_emul.c

```diff
@@ -245,27 +238,22 @@ static void rtc_emul_invoke_update_callback(const struct device *dev)
 }
 #endif /* CONFIG_RTC_UPDATE */
 
-static void rtc_emul_update(struct k_work *work)
+static void rtc_emul_update(struct k_timer *timer)
 {
-	struct rtc_emul_work_delayable *work_delayable = (struct rtc_emul_work_delayable *)work;
-	const struct device *dev = work_delayable->dev;
+	const struct device *dev = timer->user_data;
 	struct rtc_emul_data *data = (struct rtc_emul_data *)dev->data;
 
-	k_work_schedule(&work_delayable->dwork, K_MSEC(1000));
-
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	rtc_emul_increment_tm(&data->datetime);
+	K_SPINLOCK_GUARDED_SECTION(&data->lock) {
+		rtc_emul_increment_tm(&data->datetime);
 
 #ifdef CONFIG_RTC_ALARM
-	rtc_emul_test_alarms(dev);
+		rtc_emul_test_alarms(dev);
```

Comment:
If possible, please remove things that are only formatting changes.

Reply:
@cfried These are actually introduced due to the additional code block around them:

before:

```
k_mutex_lock()
rtc_emul...()
k_mutex_unlock()
```
now

```
K_SPINLOCK_GUARDED_SECTION {
      rtc_emul...()
}
```

It's just difficult to see because of the preprocessor statements.

So I believe that in this case the formatting change is inevitable (as in the other newly introduced blocks). Do you agree?